packages/input-text/src/protectedValue.js

```diff
@@ -124,19 +126,37 @@ export default WrappedComponent => {
             focused={this.state.isEditing}
             value={this.state.isEditing ? "" : this.state.value}
             onKeyDown={covered ? this.onKeyDown : onKeyDown}
+            onChange={event =>
+              this.setState({ editedValue: event.target.value })
+            }
+            onBlur={() => {
+              this.state.isEditing && this.onCancel();
+            }}
           />
           {covered && !this.state.isEditing && this._renderCover()}
           {covered &&
             isEditing && (
               <div>
                 <Button
+                  id="cancel-button"
                   onClick={this.onCancel}
                   size={buttonSizes.MEDIUM}
                   className="margin-button"
                 >
                   Cancel
                 </Button>
-                <Button type={buttonTypes.SECONDARY} size={buttonSizes.MEDIUM}>
+                <Button
+                  id="save-button"
+                  onClick={() => {
+                    this.onReplace(this.state.editedValue);
+                  }}
+                  onMouseDown={e => {
+                    // We do this in order to avoid onBlur event on input
+                    e.preventDefault();
```

Comment:
dont we want blur on save?

Reply:
That's because it cancels the edit on blur. So clicking the save button will trigger the cancel(onBlur) before saving

Reply:
we are preventing this blur https://github.com/CraveFood/farmblocks/pull/439/files#diff-4069488936b63491eb97f9c77d8a15baR140

Reply:
ah, yes, ok